src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -29,6 +32,8 @@ internal sealed partial class VisualStudioProjectTracker : ForegroundThreadAffin
         #region Mutable fields accessed only from foreground thread - don't need locking for access (all accessing methods must have AssertIsForeground).
         private readonly List<WorkspaceHostState> _workspaceHosts;
 
+        private readonly HostWorkspaceServices _workspaceServices;
```

Comment:
This is odd because the project trackers potentially references to more than one workspace, but an instance of HostWorkspaceServices is tied to a specific workspace instance.